app/stac_api/pagination.py

```diff
@@ -0,0 +1,150 @@
+            logger.error(
+                'Invalid query parameter limit=%d: negative number not allowed',
+                page_size,
+                extra={'request': request}
+            )
+            raise ValidationError(
+                _('limit query parameter to small, must be in range 1..%d') % (self.max_page_size),
+                code='limit'
+            )
+        if self.max_page_size and page_size > self.max_page_size:
+            logger.error(
+                'Invalid query parameter limit=%d: number bigger than the max size of %d',
+                page_size,
+                self.max_page_size,
+                extra={'request': request}
+            )
+            raise ValidationError(
+                _('limit query parameter to big, must be in range 1..%d') % (self.max_page_size),
+                code='limit'
+            )
+
+        return page_size
+
+    def get_raw_page_size(self, request):
+        return request.query_params.get(self.page_size_query_param, str(self.page_size))
+
+
+class GetPostCursorPagination(CursorPagination):
+
+    def get_raw_page_size(self, request):
+        if request.method == 'POST':
+            # For POST method the page size, aka `limit` parameter is in the body and not in the
+            # URL query
+            return request.data.get(self.page_size_query_param, str(self.page_size))
+        return super().get_raw_page_size(request)
+
+    def decode_cursor(self, request):
+        if request.method == 'POST':
+            # Patched the cursor in the url query from POST payload
+            cursor_encoded = request.data.get(self.cursor_query_param)
+            if cursor_encoded:
+                # Here we need to patch the request with the cursor. The original
+                # decode_cursor() method is taking the cursor from the URL query
+                # that's why we create a new request object being a copy of the
+                # original one plus the cursor as URL query. We need to do this
+                # copy because request objects are immutable.
+                environ = request.environ.copy()
+                environ['QUERY_STRING'] += parse.urlencode(
+                    {self.cursor_query_param: cursor_encoded},
+                    doseq=True,
+                )
+                request = Request(WSGIRequest(environ))
+        return super().decode_cursor(request)
+
+    def get_next_link(self, request=None):
+        next_link = super().get_next_link(request)
+        return self.patch_link(next_link, request)
+
+    def get_previous_link(self, request=None):
+        previous_link = super().get_previous_link(request)
+        return self.patch_link(previous_link, request)
+
+    def patch_link(self, link, request):
+        if link and request and request.method == 'POST':
+            cursor = get_query_param(link['href'], self.cursor_query_param)
+            limit = get_query_param(link['href'], self.page_size_query_param)
```

Comment:
Not sure, if it would have a significant influence on performance, just a thought:
Would it make sense, to make `get_query_param` accept a list of keys instead of a single key?
Then you'd only need one call here, with [`cursor`, `limit`] and in `get_query_param()` the parsing of the url and the query would only need to be done once.

Reply:
Good idea I'll modify the function as follow:
```python
def get_query_params(url, keys):
    '''Get URL query parameters by keys

    Args:
        url: string
            url to parse and retrieve query parameter
        keys: string | [string]
            query parameter key(s) to retrieve
    Returns: string | [string]
        Query parameter value(s)
    '''
    (scheme, netloc, path, query, fragment) = parse.urlsplit(url)
    query_dict = parse.parse_qs(query, keep_blank_values=True)
    if isinstance(keys, str):
        return query_dict.get(keys, None)
    return [query_dict.get(key, None) for key in keys]
```